lamindb/_curate.py

```diff
@@ -1058,6 +1062,490 @@ def save_artifact(
         return self._artifact
 
 
+def _maybe_validation_error(nonval_keys: list[str], name: str):
+    if (n := len(nonval_keys)) > 0:
+        s = "s" if n > 1 else ""
+        are = "are" if n > 1 else "is"
+        raise ValidationError(
+            f"the following {n} key{s} passed to {name} {are} not allowed: {colors.yellow(_print_values(nonval_keys))}"
+        )
+
+
+class SOMACurator(BaseCurator):
+    """Curation flow for ``tiledbsoma``.
+
+    See also :class:`~lamindb.Curator`.
+
+    Args:
+        experiment_uri: The uri of a `tiledbsoma.Experiment`.
+        var_index: The registry fields for mapping the ``.var`` indices for measurements.
+            Should be in the form `{"measurement name": ("var column", field)}`.
+            These keys should be used in the flattened form ('{measurement name}__{column name in .var}')
+            in `.standardize` or `.add_new_from`, see the output of `.var_index`.
+        categoricals: A dictionary mapping ``.obs`` columns to a registry field.
+        obs_columns: The registry field for mapping the ``.obs`` columns.
+        using_key: A reference LaminDB instance.
+        organism: The organism name.
+        sources: A dictionary mapping ``.obs`` columns to Source records.
+        exclude: A dictionary mapping column names to values to exclude.
```

Comment:
Which column names? Exclude from what?

(I guess I'm touching on broader questions in API design for Curator but given this is a new interface we should keep it as minimal as possible to just enable minimal functionality rather than propagating bad design choices.)

Reply:
This is a hack for `cellxgene` and I've already complained many times that we need it. We need a way to exclude values from being validated against that are benign "default" values.

Example:

CxG pins ontology A against version B. However, if the corresponding column is not present in the object, we need to add it with default values such as "unknown". These values such as "unknown" would then lead to `validation` failures though because they are not found in ontology A of version B. The `exclude` parameter allows us to exclude specific values from validation.

We can pick the conversation up on slack again because I consider this a hack and not a solution. For now, we need it.

Reply:
@Koncopd could you slightly improve all "exclude docstrings" given my explanation (if it was clear enough) in all places where it's used, please?

Reply:
Could you just give me a docstring to paste?